src/SessionTracker.php

```diff
@@ -281,27 +286,36 @@ protected function constructPayload($sessions)
     protected function deliverSessions()
     {
         $sessions = $this->getSessionCounts();
+
         $this->setSessionCounts([]);
-        if (count($sessions) == 0) {
+
+        if (!is_array($sessions) || count($sessions) === 0) {
```

Comment:
This `!is_array` check fixes the issue. It doesn't seem worth adding `Countable` support here, but we can do if that's desirable

Reply:
Actually we can't support `Countable` — we call `array_key_exists` elsewhere so we explicitly only support arrays here

Reply:
It would help to add phpdoc with type information to all the methods in this class, so the return type of `getSessionCounts` is clear. That function should probably throw away anything that is not null or an array, already.

Reply:
I've made this change in https://github.com/bugsnag/bugsnag-php/pull/579/commits/d2daba09ecd4d881a16e10291ae5304e57751153

I think it's safe because it's essentially the same as what we were doing internally anyway, but `getCurrentSession` is public so it could be a BC break. I'm not convinced it is though, given the bug we're fixing here is caused by returning the wrong type from the custom functions, so if anyone was relying on that it's likely they were triggering this bug (or a similar one with `lastSent`) anyway